### ISSUE TITLE
Rolling build test failure: System.Private.Uri.Unit.Tests - iOSSimulator-Release-x64-Mono_Release-OSX

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#67716]
- Rolling build (`runtime-extra-platforms`): https://dev.azure.com/dnceng/public/_build/results?buildId=1702568&view=results
- Pipeline: Build iOSSimulator x64 Release AllSubsets_Mono
- Job: https://dev.azure.com/dnceng/public/_build/results?buildId=1702568&view=logs&j=accacb66-0a14-52ef-4079-6344686d9d4c&t=34fbd35e-866b-51a1-9063-fb50703b1957&l=58
- Artifacts: https://dev.azure.com/dnceng/public/_build/results?buildId=1702568&view=ms.vss-test-web.build-test-results-tab&runId=46398566&resultId=177168&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

The [console.5254d828.log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-aee1a179460a4cd280/System.Private.Uri.Unit.Tests/1/console.5254d828.log?sv=2019-07-07&se=2022-04-26T10%3A35%3A24Z&sr=c&sp=rl&sig=hNHKQ0urUSxFy8CH1a6LLGJrEfJvzCNH5ZuiEC2XJ6s%3D) file indicates the test was cancelled, but nothing else:
```
XHarness command issued: apple test --app /tmp/helix/working/B4C3099A/w/B5B9096D/e/System.Private.Uri.Unit.Tests.app --output-directory /tmp/helix/working/B4C3099A/w/B5B9096D/uploads --target ios-simulator-64 --timeout 00:30:00 --xcode /Applications/Xcode115.app -v --launch-timeout 00:03:00
[03:36:48] info: Preparing run for ios-simulator-64
[03:36:49] info: Looking for available ios-simulator-64 simulators..
[03:36:49] dbug: Looking for available ios-simulator-64 simulators. Storing logs into list-ios-simulator-64-20220406_033649.log
[03:40:11] info: Found simulator device 'iPhone X (iOS 13.5) - created by XHarness'
[03:40:11] info: Getting app bundle information from '/tmp/helix/working/B4C3099A/w/B5B9096D/e/System.Private.Uri.Unit.Tests.app'..
[03:40:11] dbug: 
[03:40:11] dbug: Running /usr/libexec/PlistBuddy -c "Print CFBundleName" /tmp/helix/working/B4C3099A/w/B5B9096D/e/System.Private.Uri.Unit.Tests.app/Info.plist
[03:40:11] dbug: Process PlistBuddy exited with 137
[03:40:11] fail: Failed to get bundle information: Killing process 723 as it was cancelled
```

The system.log file in the artifacts (which weighs 1 MB) shows a bunch of sparse failures (modprobe, trust evaluate, flock lock maps file, ...). Not sure how or if they relate to the test failure.

I couldn't find a log file with information about the test execution.

@steveisok feel free to transfer to core-eng if needed, or mark this as duplicate of other issue if you find it.

**Comment by steveisok** [comment]
There's a 3 minute launch timeout and from the looks of the logs, it took 3 minutes to actually find the simulator.  

@carlossanlop Can you ask FR to take a look at and bounce `dci-mac-build-148.local`?

**Comment by premun** [comment]
As @steveisok says, this is really unfortunate timing of things.

I think this is a bug in XHarness where it doesn't recognize that it is the launch timeout cancellation happening and not failing operations (finding device, reading app bundle info...).

I have logged this: https://github.com/dotnet/xharness/issues/855

This should result in some exit code that will be retried and the machine rebooted which will most definitely heal it.

**Comment by premun** [comment]
This has been fixed in dotnet/xharness#859 so bumping XHarness in dotnet/runtime should resolve this (a work item retry will happen)
